deployment/frontend/src/components/dashboard/teams/forms/TeamForm.tsx

```diff
@@ -64,10 +66,51 @@ export default function TeamForm({
     const {
         register,
         setValue,
+        setError,
         watch,
         formState: { errors, isSubmitting },
     } = formObj
+    const { data: session } = useSession()
+    const username = session?.user?.name ?? ''
+    const sysadmin = session?.user?.sysadmin ?? false
     const possibleParents = api.teams.getAllTeams.useQuery()
+
+    useEffect(() => {
```

Comment:
Can we please not use useEffect on this? Also not access the DOM directly using querySelector, this is very error prone, you should probably set this on the schema level by using refine or superRefine

Reply:
If we use schema-level validation, how will it access the possibleParents.data object needed to get the capacity?

Or are you suggesting that we add capacity as a hidden form field that updates based on the selected parent, allowing us to use refine?

Also, this approach would only validate on submit, so there would be no need to disable the button.

is this what you are suggesting?

Reply:
Step 1.

Create a function that takes in the possible parent + The values that are inputed

Step 2.

Move the call to get the parents one level up to CreateTeamForm and EditTeamForm, TeamForm now has parents as a prop

Step 3

Do something like this (Dummy code)
![image](https://github.com/user-attachments/assets/5c60b535-23d5-44ed-9fca-24883e914965)


I will approve the PR, but i believe that useEffect and direct DOM manipulation is not something that the react team approves of